### PR TITLE
Bug/tokenclaimrefresh

### LINKED ISSUE #224: Wait for User to load Wallet before attempting to call subgraph
## Detailed Description
The component needs to wait for the user to allow access to their wallet before attempting to call the subgraph. It should also reload upon wallet account changes. This should also coincide with replacing the etherjs wallet account call with web3react library.

### PR DESCRIPTION
**Pull request by hhamud**
---
name: Pull Request
about: Create a pull request for this project

---

## Description
* Closes #224 
* Fixed an issue where the token claim cards were displaying individual purchases rather than a single aggregated purchase for each sale. 

## How Has This Been Tested?
 Manually Tested

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/views/Token/index.tsx [hunk 1]
```diff
@@ -1,9 +1,11 @@
 // External
 import React, { useState, useEffect } from 'react'
 import dayjs from 'dayjs'
+import { useWeb3React } from '@web3-react/core'
 import { BigNumber } from 'ethers'
 import { useTranslation } from 'react-i18next'
 import { useDispatch } from 'react-redux'
+import _ from 'lodash'
```
Comment (rossneilson) [inline src/views/Token/index.tsx:8]: Lodash is something we should really try best not to use in the frontend. It has a huge package size that really slows down page loads.
If we have a legitimate use for it then its fine but I think what is being done here can be done with vanilla es6 easily enough.

Comment (adamazad) [reply src/views/Token/index.tsx:8]: Good point. Import only the functions you need to use via the [`es` package](https://www.npmjs.com/package/lodash-es)

Comment (hhamud) [reply src/views/Token/index.tsx:8]: I didn't know about lodash slowing down pages. I liked its methods however as it is a problem, I'll remove it and use classic ES6 functions instead.

File: src/views/Token/index.tsx [hunk 2]
```diff
@@ -65,45 +67,44 @@ query getTokenClaims {
...
+              sale: purchase[0].sale,
+              amount: _.reduce(
+                purchase,
+                (total, n) => {
+                  return BigNumber.from(total).add(BigNumber.from(n.amount))
+                },
+                BigNumber.from(0)
+              ),
+            }))
+            .value()
```
Comment (adamazad) [inline src/views/Token/index.tsx:107]: The `FixedPriceSaleUser` can have totalPurchaseAmount` to aggregate this. `groupBy` is also easy to implement with reduce and `Record` type. Remove `lodash`

File: package.json [hunk 3]
```diff
@@ -108,6 +108,7 @@
     "@types/d3": "^6.3.0",
     "@types/express": "^4.17.11",
     "@types/graphql": "^14.5.0",
+    "@types/lodash": "^4.14.170",
```
Comment (rossneilson) [inline package.json:111]: I think this can be removed now

**Pull request merged**